components/raftstore/src/store/fsm/peer.rs

```diff
@@ -2984,6 +2996,12 @@ where
                     self.on_gc_peer_request(msg);
                 }
             }
+            ExtraMessageType::MsgFetchCommittedIndexRequest => {
```

Comment:
Can we achieve this without introducing new message type and response.

Reply:
Nope.

As discussing offline with @glorv, none of the current `Msgxxx` messages include the "committed_index" of this raft group. An alternative approach is to modify the internal context within `MsgHeartbeat`, so that the leader's response to followers includes the "committed_index" of this raft group.

However, both of these methods require changes to the `kvproto`. And especially, the modification to `MsgHeartbeat` should contain the changes on `raft-rs`. To keep things simple, this implementation introduces a single message pair, `MsgFetchCommittedIndexRequest` and `MsgFetchCommittedIndexResponse`, to address this issue.

Reply:
Actually, MsgAppend contains the commited_index, so maybe it's better to also set `last_leader_committed_idx ` when receive MsgAppend. This can help avoid sending the extra  MsgFetchCommittedIndexRequest.

Reply:
yep. But if the region has not updates, it won't send `MsgAppend` to the related peer, and this peer cannot get the latest `commited_index`. That's another reason why it needs `MsgFetchCommittedIndexRequest` to get the latest committed_index.

As it can avoid sending redundant `MsgFetchCommittedIndexRequest` to the leader after this peer already received `MsgAppend`,  reducing some minor costs, it can be added.

Reply:
@LykxSassinator Why not use ReadIndex request to get the lastest commit index.
components/raftstore/src/store/peer.rs in step(), it set commit index as `index` of MsgReadIndexResp
```
  let index = self.get_store().commit_index();
            // Check if the log term of this index is equal to current term, if so,
            // this index can be used to reply the read index request if the leader holds
            // the lease. Please also take a look at raft-rs.
            if self.get_store().term(index).unwrap() == self.term() {
                let state = self.inspect_lease();
                if let LeaseState::Valid = state {
                    // If current peer has valid lease, then we could handle the
                    // request directly, rather than send a heartbeat to check quorum.
                    let mut resp = eraftpb::Message::default();
                    resp.set_msg_type(MessageType::MsgReadIndexResp);
                    resp.term = self.term();
                    resp.to = m.from;

                    resp.index = index;
                    resp.set_entries(m.take_entries());

                    self.raft_group.raft.msgs.push(resp);
                    return Ok(());
                }
                self.should_wake_up = state == LeaseState::Expired;
            }
```

Reply:
Accept.